braindecode/datautil/xy.py

```diff
@@ -26,39 +24,62 @@ def create_from_X_y(
     sfreq: common sampling frequency of all trials
     ch_names: array-like
         channel names of the trials
-    drop_last_window: bool
-        whether or not have a last overlapping window, when
-        windows/windows do not equally divide the continuous signal
-    window_size_samples: int
-        window size
-    window_stride_samples: int
-        stride between windows
 
     Returns
     -------
-    windows_datasets: BaseConcatDataset
+    base_datasets: BaseConcatDataset
         X and y transformed to a dataset format that is compatible with skorch
         and braindecode
     """
-    n_samples_per_x = []
     base_datasets = []
     if sfreq is None:
-        sfreq = 100
-        log.info("No sampling frequency given, set to 100 Hz.")
+        log.error("No sampling frequency given.")
     if ch_names is None:
         ch_names = [str(i) for i in range(X.shape[1])]
         log.info(f"No channel names given, set to 0-{X.shape[1]}).")
 
-
     for x, target in zip(X, y):
-        n_samples_per_x.append(x.shape[1])
         info = mne.create_info(ch_names=ch_names, sfreq=sfreq)
         raw = mne.io.RawArray(x, info)
         base_dataset = BaseDataset(raw, pd.Series({"target": target}),
                                    target_name="target")
         base_datasets.append(base_dataset)
     base_datasets = BaseConcatDataset(base_datasets)
+    return base_datasets
+
+
+def create_windows_from_X_y(X, y, drop_last_window, sfreq=None, ch_names=None,
+        window_size_samples=None, window_stride_samples=None):
+    """Create a BaseConcatDataset of WindowsDatasets from X and y to be used for
+    decoding with skorch and braindecode, where X is a list of pre-cut trials
+    and y are corresponding targets.
+
+    Parameters
+    ----------
+    X: array-like
+        list of pre-cut trials as n_trials x n_channels x n_times
+    y: array-like
+        targets corresponding to the trials
+    sfreq: common sampling frequency of all trials
+    ch_names: array-like
+        channel names of the trials
+    drop_last_window: bool
+        whether or not have a last overlapping window, when
+        windows/windows do not equally divide the continuous signal
+    window_size_samples: int
+        window size
+    window_stride_samples: int
+        stride between windows
 
+    Returns
+    -------
+    windows_datasets: BaseConcatDataset
```

Comment:
why windows_datasets of type BaseConcatDataset and not WindowsDataset?

Reply:
So probably naming is the problem here. In braindecode, a ```BaseDataset``` is a wrapper around a single ```mne.Raw```. Similarly, a ```WindowsDataset``` is a wrapper around a single ```mne.Epochs```. Both can be concatenated using the ```BaseConcatDataset``` class. So, since here multiple ```WindowsDatasets``` are returned, it is actually a ```BaseConcatDataset``` of ```WindowsDataset```.

Reply:
thanks for the explanation. In mne when you concat raw you still get raw and if you concat epochs you still  get epochs. Is it really necessary to have another type of dataset in the public API?

Reply:
For the dataset part we tried to stick to the PyTorch API as closely as possible. Our ```BaseDataset``` inherits from ```torch.utils.data.dataset.Dataset``` and our ```BaseConcatDataset``` inherits from ```torch.utils.data.dataset.ConcatDataset```, see https://pytorch.org/docs/stable/data.html#torch.utils.data.ConcatDataset. But we can always discuss with @robintibor.

Reply:
ok then I would just propose to update the docstring:

```suggestion
    concat_dataset: BaseConcatDataset
```